src/main/webapp/2Dscript.js

```diff
@@ -80,7 +180,410 @@ function addCenterControl(controlDiv) {
+let priorLat;
+let priorLng;
+// Store whether or not selection has been clicked.
+let isMouseDown = false;
+// The markers that are generated by selection window.
+let markerBottom;
+let markerTop;
+// The part of the main path that is selected.
+let subPath;
+let infoWindow;
+const LEFTCLICK = 1;
+$(function() {
+  $(document).on({
+    mousedown: function(event) {
+      console.log('event occurred');
+      // Inital selection to draw the box.
+      if (event.ctrlKey) {
+        if (!isMouseDown) {
+          console.log('mousedown triggered');
+          placeRectangleStart();
+        }
+      } else {
+        // Place the box.
+        if (event.which === LEFTCLICK) {
+          if (isMouseDown) {
+            placeRectangleEnd();
+          }
+        }
+      }
+    },
+    mousemove: genChangingBox,
   });
+});
+
+/**
+ * Locks the starting cursor when the user begins using the selection rectangle.
+ */
+function placeRectangleStart() {
+  x = event.pageX;
+  y = event.pageY;
+  priorLat = currentLat;
+  priorLng = currentLng;
+  $('body').append('<div class="widget" style="top:' +
+    y + 'px; left: ' + x + 'px;"></div>');
+  widget = $('.widget').last();
+  isMouseDown = true;
+}
+
+/**
+ * Finish placing the selection box, select the subsection within this region.
+ */
+function placeRectangleEnd() {
+  $('.widget').remove();
+
+  const mapRuns = Object.entries(runs);
+  // Clear pop up window.
+  if (infoWindow !== undefined) {
+    infoWindow.setMap(null);
+  }
+  // Clear other paths.
+
+  for ([id, currentRun] of mapRuns) {
+    const {subLine, discoveredMinLat, discoveredMinLatPair,
+      discoveredMaxLng, discoveredMaxLngPair} = computeSubSection(
+        currentRun.data, currentLat, priorLat, currentLng, priorLng);
+    // Choose a subSectionNumber, implement differently in future pr.
+    const subSectionNumber = 1;
+    // Clear prior paths, only display newly selected ones.
+    if (markerBottom !== undefined) {
+      markerBottom.setMap(null);
+    }
+    if (markerTop !== undefined) {
+      markerTop.setMap(null);
+    }
+    if (subPath !== undefined) {
+      subPath.setMap(null);
+    }
+    subPath = getPolyLine(subLine, 'blue', 1.0, 2);
+    // Setup event listener to show option for 3D window when polyline is
+    // clicked.
+    google.maps.event.addListener(subPath, 'click', function(event) {
+      const latLng = event.latLng;
+      console.log('Polyline clicked at lat: ' + latLng.lat() +
+        ' lng: ' + latLng.lng());
+      infoWindow = new google.maps.InfoWindow({
+        content: '<a href=/home.html?id=' + id + '&dataType=' +
+          type + '&subSection=' + subSectionNumber +
+          '&stored=true' + '> View in 3D </a>',
+        position: latLng,
+      });
+      infoWindow.setMap(map);
+    },
+    );
+
+    subPath.setMap(map);
+
+    markerBottom = genMarker(discoveredMinLat, discoveredMinLatPair);
+    markerTop = genMarker(discoveredMaxLngPair, discoveredMaxLng);
```

Comment:
It seems odd to reassign `markerTop` and `markerBottom` on each iteration of the loop. What's the purpose?

Reply:
In this case it doesn't really make sense to have the loop, this is a remnant for trying to add an additional feature, I'll remove.